benchmarks/off-policy/README.md

```diff
@@ -76,216 +92,560 @@ After the experiment is finished, you can use the following command to generate
 
-### SAC
+#### SAC
 
 |      Hyperparameter      | Value |
 | :----------------------: | :---: |
-|        ``alpha``         |  0.2  |
-| ``start_learning_steps`` | 5000  |
+|        `alpha`         |  0.2  |
+| `start_learning_steps` | 5000  |
+
+### Lagragian
 
+The lagrangian versions of off-policy algorithms share the same set of
+lagrangian hyperparameters. The hyperparameters are listed below:
+
+|      Hyperparameter      | Value |
+| :----------------------: | :---: |
+| `cost_limit` | 25.0  |
+| `lagrangian_multiplier_init` | 0.001  |
+| `lambda_lr` | 0.00001  |
+| `lambda_optimizer` | Adam |
 
 ### Some Hints
 
-In our experiments, we found that somehyperparameters are important for the performance of the algorithm:
+In our experiments, we found that some hyperparameters are important for the performance of the algorithm:
 
-- ``obs_normlize``: Whether to normalize the observation.
-- ``rew_normlize``: Whether to normalize the reward.
-- ``cost_normlize``: Whether to normalize the cost.
+- `obs_normalize`: Whether to normalize the observation.
+- `reward_normalize`: Whether to normalize the reward.
+- `cost_normalize`: Whether to normalize the cost.
 
-We have done some experiments to show the effect of these hyperparameters, and we log the best configuration for each algorithm in each environment. You can check it in the ``omnisafe/configs/off_policy``.
+We have done some experiments to show the effect of these hyperparameters, and we log the best configuration for each algorithm in each environment. You can check it in the `omnisafe/configs/off_policy`.
 
-Besides, the hyperparameter ``train_cfgs:torch_num_threads`` is also important. Off-policy algorithms always use more time to update policy than to sample data. So we use ``torch_num_threads`` to speed up the update process.
+Generally, we recommend:
 
-This hyperparamter depens on the number of CPU cores. We set it to 8 in our experiments. You can set it to some other porper value according to your CPU cores.
+|      Hyperparameter      | Value |
+| :----------------------: | :---: |
+| `obs_normalize`   |  `False`  |
+| `reward_normalize`|  `False`  |
+| `cost_normalize`  | `True` |
+
+Besides, the hyperparameter `torch_num_threads` in `train_cfgs` is also important. In a single training session, a larger value for `torch_num_threads` often means faster training speed. However, we found in experiments that setting `torch_num_threads` too high can cause resource contention between parallel training sessions, resulting in slower overall experiment speed. In the configs file, we set the default value for `torch_num_threads` to 16, which ensures faster training speed for a single session. If you need to launch multiple training sessions in parallel, please consider your device configuration. For example, if your computing resources have a total of 32 physical cores and you are launching 16 training scripts in parallel, do not set `torch_num_threads` to a value greater than 4.
```

Comment:
How to understand the example at last?

Reply:
A computer with 32 physical cores has 64 thread cores. This example aims to illustrate that `the number of experiments` * `torch_num_threads` should not exceed the number of thread cores.

Reply:
`A computer with 32 physical cores has 64 thread cores` It is machine-specific, not always the case.

Reply:
I update a better version below; you can check it out.